primitives/state-machine/src/lib.rs

```diff
@@ -497,8 +487,12 @@ where
 	Exec: CodeExecutor,
 {
 	let proving_backend = proving_backend::ProvingBackend::new(trie_backend);
+	let mut extensions = Extensions::new();
+	if let Some(keystore) = keystore {
```

Comment:
Do we really need `keystore` here? If that can be invoked from the network context (i.e. light client requesting to prove some execution) we can end up signing something with a local key and giving it back as a response.